tests/api/printf.cpp

```diff
@@ -128,17 +160,12 @@ TEST_F(WithCommandQueue, SimplePrintf) {
     ASSERT_STREQ(printf_buffer, message);
 }
 
-TEST_F(WithCommandQueue, TooLongPrintf) {
-    // each print takes 12 bytes (4 for the printf_id, and 2*4 for the 2 integer
-    // to print) + 4 for the byte written counter
-    auto cfg1 =
-        CLVK_CONFIG_SCOPED_OVERRIDE(printf_buffer_size, uint32_t, 28, true);
-
-    temp_folder_deletion temp;
-    stdoutFileName = getStdoutFileName(temp);
-
-    int fd;
-    ASSERT_TRUE(getStdout(fd));
+TEST_F(WithPrintfEnabled, TooLongPrintf) {
+    auto props = setup_arm_printf_test(44);
+    SetUpWithContextProperties(props.data());
```

Comment:
`SetUpWithContextProperties` is always called after `setup_arm_printf_test`. We should move it inside to avoid duplication.
Also we need to call `WithPrintfEnabled::TearDown()` before calling `SetUpWithContextProperties` to make sure that everything has been deleted before creating the new context. You can check that part using valgrind, to make sure there is no memory leak.

Reply:
I tried that previously (to move the `SetUpWithContextProperties` inside the func(I am assuming that is what you mean) but that was failing as it does not have the scope for the `WithPrintfEnabled`